core/bot_framework_adapter_test.go

```diff
@@ -67,15 +75,37 @@ func Example() {
 			ID:   "1234abcd",
 			Name: "SteveW",
 		},
-		Text:      "Message from Teams Client",
-		ReplyToID: "5d5cdc723",
+		Text:       "Message from Teams Client",
+		ReplyToID:  "5d5cdc723",
+		ServiceURL: srv.URL,
 	}
 
-	// Pass the activity and handler to the adapter for proecssing
-	ctx := context.Background()
-	err = adapter.ProcessActivity(ctx, activity, customHandler)
-	if err != nil {
-		fmt.Println("Failed to process request", err)
-		return
-	}
+	handler := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		ctx := context.Background()
+		setting := core.AdapterSetting{
+			AppID:       "asdasd",
+			AppPassword: "cfg.MicrosoftTeams.AppPassword",
+		}
+		setting.CredentialProvider = auth.SimpleCredentialProvider{
+			AppID:    setting.AppID,
+			Password: setting.AppPassword,
+		}
+		clientConfig, err := client.NewClientConfig(setting.CredentialProvider, auth.ToChannelFromBotLoginURL[0])
+		assert.Nil(t, err, fmt.Sprintf("Failed with error %s", err))
+		connectorClient, err := client.NewClient(clientConfig)
+		assert.Nil(t, err, fmt.Sprintf("Failed with error %s", err))
+		adapter := core.BotFrameworkAdapter{setting, &core.MockTokenValidator{}, connectorClient}
+		act, err := adapter.ParseRequest(ctx, req)
+		assert.Nil(t, err, fmt.Sprintf("Failed with error %s", err))
+		err = adapter.ProcessActivity(ctx, act, customHandler)
+		assert.Nil(t, err, fmt.Sprintf("Failed with error %s", err))
+	})
+	rr := httptest.NewRecorder()
+	bodyJSON, _ := json.Marshal(activity)
```

Comment:
Please handle the error returned. I promise this is the change I am requesting :see_no_evil: 
Thanks for being patient.

Reply:
No problem. Quality is the priority.